gpu_ocean/SWESimulators/gpu_kernels/KP07_kernel.cu

```diff
@@ -293,75 +371,97 @@ __global__ void swe_2D(
    
     // Read H into sheared memory
     readBlock2single(Hi_ptr_, Hi_pitch_,
-		     Hi, nx_, ny_);
-    __syncthreads();
-
-    // Reconstruct H at the cell faces
-    reconstructH(RHx, RHy, Hi);
+                     Hi, nx_, ny_);
     __syncthreads();
-
+    
     //Fix boundary conditions
     // TODO: Add if on boundary condition
     if (bc_north_ == 1 || bc_east_ == 1 || bc_south_ == 1 || bc_west_ == 1)
     {
-	noFlowBoundary2Mix(Q, nx_, ny_, bc_north_, bc_east_, bc_south_, bc_west_);
+        noFlowBoundary2Mix(Q, nx_, ny_, bc_north_, bc_east_, bc_south_, bc_west_);
         __syncthreads();	
-	// Looks scary to have fence within if, but the bc parameters are static between threads.
+        // Looks scary to have fence within if, but the bc parameters are static between threads.
     }
     
+    // Reconstruct Q in x-direction into Qx
+    
     //Reconstruct slopes along x and axis
     // The Qx is here dQ/dx*0.5*dx
     minmodSlopeX(Q, Qx, theta_);
-    minmodSlopeY(Q, Qy, theta_);
     __syncthreads();
 
     // Adjust the slopes to avoid negative values at integration points
-    adjustSlopes(Qx, Qy, RHx, RHy, Q);
-    __syncthreads();
+    //adjustSlopes_x(Qx, Hi, Q);
```

Comment:
Dead code.

Reply:
Well spotted! This code should of course be included again. It was commented out for the purpose of fair performance benchmarking. Fixed now :+1: